File: app/src/main/java/xyz/hasnat/csdictionary/DicLoadFromFile.java
package xyz.hasnat.csdictionary;

import java.io.BufferedReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

import xyz.hasnat.csdictionary.Model.Word;

public class DicLoadFromFile {
    /**
     *  For first time DATABASE installation,
     * @param bufferedReader
     * @param databaseHelper
     */
    public static void loadData(BufferedReader bufferedReader, DicDatabaseHelper databaseHelper){
        ArrayList<Word> wordArrayList = new ArrayList<>();

        try {
            BufferedReader fileReader = bufferedReader;
            try {
                int c = fileReader.read();
                while (c != (-1)) {
                    StringBuilder stringBuilder = new StringBuilder();
                    while ((char)c != '\n' && c != -1) {
                        try {
                            stringBuilder.append((char)c);
                        } catch (Exception e) {
                            System.out.println(stringBuilder.length());
                            //e.printStackTrace();
                        }
                        c = fileReader.read();
                        if (c == -1) {
                            return;
                        }
                    }
                    String wordString = stringBuilder.toString(); // only word


                    // passing definition
                    ArrayList<String> definition = new ArrayList<>();
                    while (c == '\n' || c == '\t') {
                        c = fileReader.read();
                        if (c == '\n' || c == '\t' || c == '\r') {
                            StringBuilder stringBuilder2 = new StringBuilder();
                            while (c != '\n') {
                                stringBuilder2.append((char)c);
                                c=fileReader.read();
                            }
                            String definitionString = stringBuilder2.toString();
                            definition.add(definitionString);
                        }else {
                            break;
                        }

                    }
                    wordString=wordString.trim();
                    //Log.e("word Load: "+(++counter)+" :"+wordString);
                    wordArrayList.add(new Word(wordString, definition)); // add to array list
                }
            } catch (IOException e) {
                e.printStackTrace();
            }

            try {
                // after adding word to array list, pass these into DATABASE for initial setup
                databaseHelper.firstTimeDatabaseInstallation(wordArrayList);
                fileReader.close();
            } catch (IOException e) {
                e.printStackTrace();
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/xyz/hasnat/csdictionary/database/DicLoadFromFile.kt
package xyz.hasnat.csdictionary.database

import java.io.BufferedReader
import java.io.IOException
import java.util.ArrayList

import xyz.hasnat.csdictionary.model.Word

object DicLoadFromFile {
    /**
     * For first time DATABASE installation,
     * @param bufferedReader
     * @param databaseHelper
     */

    fun loadData(bufferedReader: BufferedReader, databaseHelper: DicDatabaseHelper) {
        val wordArrayList = ArrayList<Word>()

        try {
            try {
                var c = bufferedReader.read()
                while (c != -1) {
                    val stringBuilder = StringBuilder()
                    while (c.toChar() != '\n' && c != -1) {
                        try {
                            stringBuilder.append(c.toChar())
                        } catch (e: Exception) {
                            println(stringBuilder.length)
                            //e.printStackTrace();
                        }

                        c = bufferedReader.read()
                        if (c == -1) {
                            return
                        }
                    }
                    var wordString = stringBuilder.toString() // only word


                    // passing definition
                    val definition = ArrayList<String>()
                    while (c == '\n'.toInt() || c == '\t'.toInt()) {
                        c = bufferedReader.read()
                        if (c == '\n'.toInt() || c == '\t'.toInt() || c == '\r'.toInt()) {
                            val stringBuilder2 = StringBuilder()
                            while (c != '\n'.toInt()) {
                                stringBuilder2.append(c.toChar())
                                c = bufferedReader.read()
                            }
                            val definitionString = stringBuilder2.toString()
                            definition.add(definitionString)
                        } else {
                            break
                        }

                    }
                    wordString = wordString.trim { it <= ' ' }
                    //Log.e("word Load: "+(++counter)+" :"+wordString);
                    wordArrayList.add(Word(wordString, definition)) // add to array list
                }
            } catch (e: IOException) {
                e.printStackTrace()
            }

            try {
                // after adding word to array list, pass these into DATABASE for initial setup
                databaseHelper.firstTimeDatabaseInstallation(wordArrayList)
                bufferedReader.close()
            } catch (e: IOException) {
                e.printStackTrace()
            }

        } catch (e: Exception) {
            e.printStackTrace()
        }

    }
}
